src/Generator/Utils/BlockGenerator.cs

```diff
@@ -50,6 +50,9 @@ public enum BlockKind
         Destructor,
         AccessSpecifier,
         Fields,
+        ConstructorBody,
```

Comment:
Do we need these explicit block kinds or could you just use the generic `MethodBody` one?

Reply:
`MethodBody` does not allow consumer to specifically know which method is being generated. These explicit blocks allow the consumer to detect when ctor, dtor or finalizer are generated.